firebase-database/src/main/java/com/google/firebase/database/core/utilities/PushIdGenerator.java

```diff
@@ -57,6 +65,49 @@ public static synchronized String generatePushChildName(long now) {
     return result.toString();
   }
 
+  public static final String successor(String key) {
+    Validation.validateNullableKey(key);
+    Integer num = tryParseInt(key);
+    if (num != null) {
+      if (Long.valueOf(num) + 1L > Integer.MAX_VALUE) {
```

Comment:
You don't need to convert `num` to long. In Java, long+int=long (and the int is treated as a long)